src/south/ocnos/goldstone/south/ocnos/interfaces.py

```diff
@@ -0,0 +1,898 @@
+from goldstone.lib.core import *
+import sysrepo
+import ncclient
+from ncclient.xml_ import *
+import libyang
+from lxml import etree
+from .util import *
+
+logger = logging.getLogger(__name__)
+
+SPEED_MAP = {
+    "SPEED_100M": "100m",
+    "SPEED_1000M": "1g",
+    "SPEED_2500M": "2.5g",
+    "SPEED_10G": "10g",
+    "SPEED_20G": "20g",
+    "SPEED_25G": "25g",
+    "SPEED_40G": "40g",
+    "SPEED_50G": "50g",
+    "SPEED_100G": "100g",
+}
+
+GOLDSTONE_INTERFACE_TOP = "/goldstone-interfaces:interfaces/interface"
+
+
+class IfChangeHandler(ChangeHandler):
+    def __init__(self, server, change):
+        super().__init__(server, change)
+        xpath = change.xpath
+
+        xpath_list = list(libyang.xpath_split(xpath))
+
+        assert xpath_list[0][0] == "goldstone-interfaces"
+        assert xpath_list[0][1] == "interfaces"
+        assert xpath_list[1][1] == "interface"
+        assert xpath_list[1][2][0][0] == "name"
+        self.ifname = xpath_list[1][2][0][1]
+
+
+class AdminStatusHandler(IfChangeHandler):
+    def apply(self, user):
+        if self.type in ["created", "modified"]:
+            value = self.change.value
+            if type(value) is dict:
+                return
+        else:
+            # admin-status is UP in default in OcNOS
```

Comment:
we should use the default value of Goldstone, not OcNOS.

When users delete the admin-status configuration, they expect that the default value of the Goldstone YANG model is used, which is `DOWN`.

https://github.com/oopt-goldstone/goldstone-mgmt/blob/master/yang/goldstone-interfaces.yang#L850